src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -1651,6 +1651,12 @@ namespace smt::noodler {
             }
             return update_model_and_aut_ass(var, result);
         } else if (solution.aut_ass.contains(var)) {
+            // as a heuristic, we check if the automaton for var is a singleton (this can sometimes help if there is a cycle in inclusions)
```

Comment:
```suggestion
            // as a heuristic, we check if the automaton for var contains exactly one word, if yes, we immediately return this word instead of going trough inclusions (this can sometimes help if there is a cycle in inclusions)
```